classes/kernel.oeclass

```diff
@@ -238,38 +239,47 @@ UIMAGE_KERNEL_OUTPUT = ""
 UIMAGE_KERNEL_OUTPUT:USE_kernel_uimage = "arch/${KERNEL_ARCH}/boot/uImage"
 KERNEL_OUTPUT += "${UIMAGE_KERNEL_OUTPUT}"
 
-CLASS_FLAGS += "kernel_dtb kernel_dtc kernel_dtc_flags kernel_dtc_source"
+CLASS_FLAGS += "kernel_dtb kernel_dtc kernel_dtc_flags kernel_dts"
 DEFAULT_USE_kernel_dtc_flags = "-R 8 -p 0x3000"
-DEFAULT_USE_kernel_dtc_source = "arch/${KERNEL_ARCH}/boot/dts/${MACHINE}.dts"
+DEFAULT_USE_kernel_dts = "arch/${KERNEL_ARCH}/boot/dts/${MACHINE}.dts"
 
+KERNEL_DTB_DIR = "arch/${KERNEL_ARCH}/boot/dts/"
 addhook kernel_devicetree_init to post_recipe_parse after set_useflags
 def kernel_devicetree_init(d):
-    kernel_dtc = d.get('USE_kernel_dtc')
     kernel_dtb = d.get('USE_kernel_dtb')
-    if kernel_dtc and kernel_dtc != 0:
-	kernel_dtc_source = d.get('USE_kernel_dtc_source')
-	dts = os.path.basename(kernel_dtc_source)
+    kernel_dtc = d.get('USE_kernel_dtc')
+    kernel_dts = d.get('USE_kernel_dts')
+    if (kernel_dtc and kernel_dts != 0) or kernel_dts != 0:
```

Comment:
maybe, but that whole function seems to try to automatically figure out how things should be set up, I honestly don't like that. How about we just have flags for all settings, and then error out if the combinations are wrong? e.g. allow the users to specify DTBDIR with a USE as well